docs/interactive_engine/tutorial_ldbc_gremlin.md

```diff
@@ -831,3 +831,1087 @@ At last, we would like to introduce `endV()` step in path expansion. Since a pat
+
+```python
+# pA and pB know each other
+# pB create a comment that reply a message created by pA
+q1 = g.execute('g.V().match(__.as(\'pA\').both(\'1..3\', \'knows\').as(\'pB\'),\
+                       __.as(\'pA\').in(\'hasCreator\').has(\'creationDate\', gt(\'2012-01-01\')).as(\'m\'),\
+                       __.as(\'pB\').in(\'hasCreator\').has(\'creationDate\', gt(\'2012-01-01\')).as(\'c\'),\
+                       __.as(\'c\').out(\'replyOf\').as(\'m\'))')
+print(q1.all())
+```
+
+The output should be like:
+
+```bash
+[{'pA': v[216172782113783812], 'pB': v[216172782113783882], 'c': v[36], 'm': v[360287970189640010]}, {'pA': v[216172782113783812], 'pB': v[216172782113783882], 'c': v[37], 'm': v[360287970189640010]}, {'pA': v[216172782113783812], 'pB': v[216172782113784105], 'c': v[38], 'm': v[360287970189640010]}, {'pA': v[216172782113783812], 'pB': v[216172782113783882], 'c': v[41], 'm': v[360287970189640010]}, {'pA': v[216172782113784105], 'pB': v[216172782113783882], 'c': v[43], 'm': v[42]}, {'pA': v[216172782113783814], 'pB': v[216172782113783962], 'c': v[50], 'm': v[360287970189640135]}, {'pA': v[216172782113783814], 'pB': v[216172782113784171], 'c': v[52], 'm': v[360287970189640135]}, {'pA': v[216172782113784481], 'pB': v[216172782113784199], 'c': v[54], 'm': v[49]}, {'pA': v[216172782113783814], 'pB': v[216172782113784038], 'c': v[56], 'm': v[360287970189640135]}, {'pA': v[216172782113783816], 'pB': v[216172782113784144], 'c': v[175], 'm': v[360287970189640462], ......, }]
+```
+
+## Advanced Techniques
+
+Then we would like to introduce some advanced techniques of gremlin steps that GIE supports. 
+
+### Filter Steps
+
+#### `hasId()`
+
+In previous tutorial, we have introduced that you can use `g.V(id)` to pick out the vertex having specific required global id from the Property Graph. But, what if we want to apply filters based on global id during the traversal? 
+
+You can use the `hasId(id)` step to achieve the purpose. For example, if you want the expanded vertex exactly having global id 144115188075861858, you can write the following codes in GIE:
+
+```python
+q1 = g.execute('g.V().out().hasId(144115188075861858)')
+print(q1)
+```
+
+The output should be like:
+
+```bash
+[v[144115188075861858], v[144115188075861858], v[144115188075861858]]
+```
+
+Note that `hasId(id)` is different from `has('id', id)`! As mentioned before, `hasId(id)` apply filter based on the global id, while `has('id', id)` apply filter based on the local id. For example, if you write:
+
+```python
+q1 = g.execute('g.V().has(\'id\', 0)')
+print(q1.all())
+```
+
+The output would be:
+
+```bash
+[v[72057594037927936],
+ v[144115188075855872],
+ v[288230376151711744],
+ v[432345564227567616],
+ v[504403158265495555]]
+```
+
+which are all the vertices whose local id is 0. 
+
+#### `where()`
```

Comment:
Can introduce the syntactic sugar of `expr()` (in supported gremlin steps) for simplifying the filtering expression.